x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/SparseFileTracker.java

```diff
@@ -255,18 +255,21 @@ private long computeLengthOfRanges() {
         }
     }
 
-    private void onGapSuccess(final long start, final long end) {
+    private boolean assertPendingRangeExists(Range range) {
+        assert Thread.holdsLock(mutex);
+        final SortedSet<Range> existingRanges = ranges.tailSet(range);
+        assert existingRanges.isEmpty() == false;
+        final Range existingRange = existingRanges.first();
+        assert existingRange.start == range.start && existingRange.end == range.end && existingRange.isPending();
```

Comment:
Can we `assert existingRange == range`? I think it should always be the very same object here.

Reply:
I think we can, yes